package.json

```diff
@@ -65,10 +66,15 @@
     "tslib": "^2.4.0",
     "undici": "^6.12.0"
   },
+  "peerDependencies": {
+    "@opentelemetry/api": "1.x",
```

Comment:
Shouldn't there be an entry in "dependencies" for `@opentelemetry/api`, as it is imported/required by the using code?

I suppose using peerDeps typically works for modern npm, as peerDeps are installed by default (https://docs.npmjs.com/cli/v10/configuring-npm/package-json#peerdependencies).  I say "typically" because some projects unfortunately use the `legacy-peer-deps` npm config option (https://docs.npmjs.com/cli/v10/using-npm/config#legacy-peer-deps) which results in peerDeps being ignored.  The result, in this case, is an `@elastic/transport` that breaks.  Here is a smallish repro of that breakage:

```
% cat package.json
{
  "name": "asdf.20240624t094529",
  "version": "1.0.0",
  "dependencies": {
    "@elastic/transport": "git+https://github.com/elastic/elastic-transport-js.git#otel"
  }
}

% npm install --legacy-peer-deps
...

% npm ls -a
npm ERR! code ELSPROBLEMS
npm ERR! invalid: @elastic/transport@8.6.0 /Users/trentm/tmp/asdf.20240624T094529/node_modules/@elastic/transport
npm ERR! missing: @opentelemetry/api@1.x, required by @elastic/transport@8.6.0
npm ERR! missing: @opentelemetry/sdk-trace-base@1.x, required by @elastic/transport@8.6.0
asdf.20240624t094529@1.0.0 /Users/trentm/tmp/asdf.20240624T094529
└─┬ @elastic/transport@8.6.0 invalid: "git+https://github.com/elastic/elastic-transport-js.git#otel" from the root project
  ├── UNMET DEPENDENCY @opentelemetry/api@1.x
  ├── UNMET DEPENDENCY @opentelemetry/sdk-trace-base@1.x
  ├─┬ debug@4.3.5
  │ └── ms@2.1.2
  ├── hpagent@1.2.0
  ├── ms@2.1.3
  ├── secure-json-parse@2.7.0
  ├── tslib@2.6.3
  └── undici@6.19.2
...
```

Here I manually copy in the built JS code from a git working copy in a separate dir:

```
% rsync -av ~/el/elastic-transport-js2/lib/ node_modules/@elastic/transport/lib/
```

Then importing the package fails:

```
% node
Welcome to Node.js v18.18.2.
Type ".help" for more information.
> require('@elastic/transport')
Uncaught Error: Cannot find module '@opentelemetry/api'
Require stack:
- /Users/trentm/tmp/asdf.20240624T094529/node_modules/@elastic/transport/lib/Transport.js
- /Users/trentm/tmp/asdf.20240624T094529/node_modules/@elastic/transport/index.js
- <repl>
    at Module._resolveFilename (node:internal/modules/cjs/loader:1077:15)
    at Module._load (node:internal/modules/cjs/loader:922:27)
    at Module.require (node:internal/modules/cjs/loader:1143:19)
    at require (node:internal/modules/cjs/helpers:119:18) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/trentm/tmp/asdf.20240624T094529/node_modules/@elastic/transport/lib/Transport.js',
    '/Users/trentm/tmp/asdf.20240624T094529/node_modules/@elastic/transport/index.js',
    '<repl>'
  ]
}
```

I don't think there is a significant downside in taking the direct dependency on `@opentelemetry/api`, is there?

Reply:
Good call. I'd initially intended to put the SDK in as a peer dependency, but forgot to clean that up. Fixed in ec782d636482749df8ec2386aa399869f7a6569a.